### PR TITLE
Update various dependencies

### PR DESCRIPTION
**Pull request by stefanscheidt**
Deliberately not yet bumping Gradle to the just released version 8, but only to the most current 7.x version (7.6.1).

Also adding .gitattributes with EOL settings for Windows cmd and bat files.

### CODE REVIEW

File: SupermarketReceipt-solution/build.gradle [hunk 1]
```diff
@@ -8,10 +8,9 @@ repositories {
 }
 
 dependencies {
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.1'
-    testImplementation 'org.junit.jupiter:junit-jupiter-params:5.8.1'
-    testImplementation 'org.junit.jupiter:junit-jupiter-engine:5.8.1'
-    testImplementation 'com.approvaltests:approvaltests:12.1.1'
+    testImplementation(platform('org.junit:junit-bom:5.9.2'))
```
Comment (stefanscheidt) [inline SupermarketReceipt-solution/build.gradle:11]: This is inspired by https://github.com/junit-team/junit5-samples/blob/main/junit5-jupiter-starter-gradle/build.gradle

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+*.so binary
+*.dll binary
+ 
+ # Behaviour for images
+*.png binary
+*.jpg binary
+*.gif binary
+
+# These are explicitly windows files and should use crlf
+*.bat text eol=crlf
```
Comment (stefanscheidt) [inline .gitattributes:30]: This helps handling `gradlew.bat`.

File: SupermarketReceipt-solution/build.gradle (hunk 1 above)
Comment (stefanscheidt) [reply SupermarketReceipt-solution/build.gradle:11]: I did the same in the other Gradle projects.

**Comment by emilybache** [comment]
Thankyou so much! This looks great

**Pull request merged**